tests/composition/test_show_graph.py

```diff
@@ -668,6 +668,106 @@ def test_of_show_nested_show_cim_and_show_node_structure_with_singleton_in_outer
+        'show_graph_kwargs, expected_output, obj_mech', test_args,
+        # ids=[str(x) for x in _nested_show_graph_kwargs]*2
+        ids=[str(x) for x in ids]
+    )
+    def test_projections_from_nested_comp_to_ocm_or_obj_mech(self, show_graph_kwargs, expected_output, obj_mech):
+        ia = ProcessingMechanism(name='ia')
+        ib = ProcessingMechanism(name='ib')
+        ic = ProcessingMechanism(name='ic')
+        id = ProcessingMechanism(name='id')
+        icomp = Composition(pathways=[ia,ib,ic,id], name='INNER COMP')
+        ocomp = Composition(nodes=[icomp], name='OUTER COMP')
+
+        # # FIX: THIS IS SKIPPED DUE TO BUG (
+        # #      ??SAME BUG AS IN MESSAGE FOR COMMIT eb61303808ad2a5ba46fdd18d0e583283397915c ??)
+        # if ('show_node_structure' in show_graph_kwargs
+        #         and not 'show_cim' in show_graph_kwargs
+        #         and ('show_nested' in show_graph_kwargs and show_graph_kwargs['show_nested']==NESTED)
+        # ):
+        #     pytest.skip("??graphviz BUG")
+
+        if obj_mech == 'monitor_for_control':
+            monitor_for_control = [ic,ib]
+            objective_mechanism = None
+        else:
+            monitor_for_control = None
+            objective_mechanism = [ic,ib]
+
+        ocm = OptimizationControlMechanism(
+            state_features=ia.input_port,
+            monitor_for_control=monitor_for_control,
+            objective_mechanism=objective_mechanism,
+            function=GridSearch(),
+            allow_probes=True,
+            control_signals=ControlSignal(modulates=(SLOPE,ia),
+                                          allocation_samples=[10, 20, 30])
+        )
+        ocomp.add_controller(ocm)
+        gv = ocomp.show_graph(output_fmt='source', **show_graph_kwargs)
+
+        try:
+            assert gv.strip() == expected_output
+        except AssertionError:
+            # ??SAME BUG AS IN MESSAGE FOR COMMIT eb61303808ad2a5ba46fdd18d0e583283397915c ??)
+            if ('show_node_structure' in show_graph_kwargs
+                    and 'show_cim' not in show_graph_kwargs
+                    and ('show_nested' in show_graph_kwargs and show_graph_kwargs['show_nested']==NESTED)
+            ):
+                # The test in this condition is for incorrect show_graph output due an as yet undetermined bug
+                #    in which the Projection from the OCM receives a projection to its control_signal output_port
+                #    from a the input_port of a nested monitored node (ic) rather than from its output_port to the
+                #    OCM's outcome_input_port.
+                # If the test fails in this condition, it could mean that the bug has been corrected.
+                # The bug may be the same one as in eb61303808ad2a5ba46fdd18d0e583283397915c
+                raise(AssertionError,"FAILURE TO REPLICATE BUGGY SHOW_GRAPH OUTPUT -- SEE COMMENT IN TEST")
+            else:
+                raise(AssertionError)
```

Comment:
thiscan be just:
`raise`

to forward the existing exception/assertion instead of creating a new one